keps/sig-api-machinery/4793-revise-apf-default-configuration/README.md

```diff
@@ -0,0 +1,992 @@
+borrowing is also instroduced to better utilize spare concurrency. The limits on
+borrowing are two-sided: a given priority level has a limit on how much it may
+borrow and a limit on how much may be borrowed from it. Worth mentioning, APF
+controller periodically populates current concurrency shares for each priority
+level (i.e. concurrency share that can be borrowed or need to be borrowed) by
+taking past usage into consideration. So in combination of those two aspects, a
+priority level's Nomial Share is only an initial value and the actual share
+might increase or decrease according to actual usage with borrowing. However,
+the owner priority level should have full Nominal Share as its current
+concurrency share if in full utilization.
+
+This KEP proposes the following settings:
+
+Name            | Current Nominal Shares | Current Lendable | Current Borrowing Limit | Current Guaranteed Shares | Proposed Nominal Shares | Proposed Lendable | Proposed Borrowing Limit | Proposed Guaranteed Shares
+--------------- | ---------------------: | ---------------: | ----------------------: | ------------------------: | ----------------------: | ----------------: | -----------------------: | -------------------------:
+exempt          | 0                      | 50%              | none                    | NA                        | 0                       | 0%                | none                     | NA
+leader-election | 10                     | 0%               | none                    | 10                        | 50                      | 80%               | none                     | 10
+node-high       | 40                     | 25%              | none                    | 30                        | 60                      | 50%               | none                     | 30
+system          | 30                     | 33%              | none                    | 20                        | 40                      | 50%               | none                     | 20
+workload-high   | 40                     | 50%              | none                    | 20                        | 40                      | 50%               | none                     | 20
+workload-low    | 100                    | 90%              | none                    | 10                        | 40                      | 75%               | none                     | 10
+global-default  | 20                     | 50%              | none                    | 10                        | 10                      | 50%               | none                     | 10
+catch-all       | 5                      | 0%               | none                    | 5                         | 5                       | 0%                | none                     | 5
+event           | NA (new)               | NA (new)         | NA (new)                | NA (new)                  | 5                       | 0%                | 100%                     | 5
```

Comment:
I'm wondering if we shouldn't adjust things in a way that SUM remains the same as it was exactly.
In other words if we provide some shares for this PL, subtract it from somewhere else.

Reply:
Also - why these shares are guaranteed? You wrote explicitly that events are best-effort, so I would expect these shares to be fully lendable.

Reply:
There are two things we can remain the same:

1. the guaranteed shares
2. the nominal shares

But we can't keep both the same. In current version, I tend to keep the guaranteed shares the same as suggested by @MikeSpreitzer because it's really difficult to not change nominal shares when we want to increase shares for more critical PLs while minimizing wastes. For example, we have to significant increase nominal shares  for leader-election, to compensate this and keep the same SUM of all nominal shares, we have to more significantly reduce the nominal shares of less critical PLs such as workload-high/low (in current KEP they are already reduced significantly so little room to further reduce).


re: events, I simply copied catch-all as I think events should have minimal guarantee at least. But I don't have strong opinion on it.